Source/Randomizer.cpp

```diff
@@ -132,16 +142,75 @@ void Randomizer::Randomize() {
     RandomizeChallenge();
 
     // Content swaps -- must happen before squarePanels
-    Randomize(upDownPanelsSetZero, SWAP::LINES | SWAP::COLORS);
-    Randomize(upDownPanelsSetOne, SWAP::LINES | SWAP::COLORS);
-    Randomize(upDownPanelsSetTwo, SWAP::LINES | SWAP::COLORS);
-    Randomize(upDownPanelsSetThree, SWAP::LINES | SWAP::COLORS);
-    Randomize(upDownPanelsSetFour, SWAP::LINES | SWAP::COLORS);
-    Randomize(leftForwardRightPanelsSetOne, SWAP::LINES | SWAP::COLORS);
-    Randomize(leftForwardRightPanelsSetTwo, SWAP::LINES | SWAP::COLORS);
-
-    Randomize(quarryLaserOptions, SWAP::LINES | SWAP::COLORS);
-    Randomize(squarePanels, SWAP::LINES | SWAP::COLORS);
+    if (_doubleRandomizer) {
+        // The pool that Tutorial Back Left is in has some panels that may not
+        // be solveable in the down position after using Sigma's randomizer. We
+        // do not want to swap any such panel with Tutorial Back Left, because
+        // that would make it impossible to exit Tutorial.
+        const std::set<int> bannedTutorialBackLeftPanels = {
+            0x00070, // Symmetry Island Fading Lines 5
+            0x01E5A, // Mill Entry Door Left
+            0x00072, // Symmetry Island Fading 3
+            0x00076, // Symmetry Island Fading 7
+            0x3C125, // Mill Control Room Extra Panel
+            0x09E85, // Tunnels Into Town Big Triangles
+        };
+        std::vector<int> upDownPanelsSetThreeDoubleMode = copyWithoutElements(upDownPanelsSetThree, bannedTutorialBackLeftPanels);
```

Comment:
If you're not re-using the variable, just inline it.
```suggestion
        std::vector<int> upDownPanelsSetThreeDoubleMode = copyWithoutElements(upDownPanelsSetThree, {
            0x00070, // Symmetry Island Fading Lines 5
            0x01E5A, // Mill Entry Door Left
            0x00072, // Symmetry Island Fading 3
            0x00076, // Symmetry Island Fading 7
            0x3C125, // Mill Control Room Extra Panel
            0x09E85, // Tunnels Into Town Big Triangles
        });
```